Sources/SwiftGRPC/Core/Channel.swift

```diff
@@ -81,4 +81,145 @@ public class Channel {
     let underlyingCall = cgrpc_channel_create_call(underlyingChannel, method, host, timeout)!
     return Call(underlyingCall: underlyingCall, owned: true, completionQueue: completionQueue)
   }
+
+  public func connectivityState(tryToConnect: Bool = false) -> ConnectivityState {
+    return ConnectivityState.connectivityState(cgrpc_channel_check_connectivity_state(underlyingChannel, tryToConnect ? 1 : 0))
+  }
+
+  public func subscribe(sourceState: ConnectivityState, tryToConnect: Bool = false, callback: @escaping (ConnectivityState) -> ()) {
```

Comment:
Would you mind adding a comment to this method, especially to explain what `sourceState` is for? It looks to me like the callback is only fired if the channel's state changes to `sourceState`, is that correct? Why not subscribe to _all_ state changes and have the caller exclude unwanted ones themselves in the callback (could be done with a simple `if state != sourceState { return }` in the callback)? That would also avoid spinning up a separate thread for each source state.